crates/shared/src/bad_token/trace_call.rs

```diff
@@ -55,13 +56,43 @@ impl TokenOwnerFinding for BalancerVaultFinder {
 pub struct UniswapV3Finder {
     pub factory: IUniswapV3Factory,
     pub base_tokens: Vec<H160>,
+    fee_values: Vec<u32>,
 }
 
 impl UniswapV3Finder {
+    pub async fn new(
+        factory: IUniswapV3Factory,
+        base_tokens: Vec<H160>,
+        current_block: u64,
+    ) -> Result<Self> {
+        // We fetch these once at start up because we don't expect them to change often.
+        // Alternatively could use a time based cache.
+        let fee_values = Self::fee_values(&factory, current_block).await?;
+        tracing::debug!(?fee_values);
+        Ok(Self {
+            factory,
+            base_tokens,
+            fee_values,
+        })
+    }
+
     // Possible fee values as given by
     // https://github.com/Uniswap/v3-core/blob/9161f9ae4aaa109f7efdff84f1df8d4bc8bfd042/contracts/UniswapV3Factory.sol#L26
-    // Could theoretically change in the future in which case hard coded values would become wrong.
-    const FEE: [u32; 3] = [500, 3000, 10000];
+    async fn fee_values(factory: &IUniswapV3Factory, current_block: u64) -> Result<Vec<u32>> {
+        // We expect there to be few of these kind of events (currently there are 4) so fetching all
+        // of them is fine. Alternatively we could index these events in the database.
+        let events = factory
+            .events()
+            .fee_amount_enabled()
+            .from_block(BlockNumber::Earliest)
+            .to_block(BlockNumber::Number(
+                current_block.saturating_sub(MAX_REORG_BLOCK_COUNT).into(),
```

Comment:
What bad could happen if we don't substract MAX_REORG_BLOCK_COUNT?

Reply:
Theoretically they could have just added a new fee constant which we load and then the block gets re-orged so that fee constant disappears. This is unlikely and the only impact is that we would waste some calls to the node. Still feels slightly nicer to subtract the constant.